app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -330,7 +340,7 @@ private void addRoomColumns(@NonNull HorizontalSnapScrollView horizontalScroller
         LecturesByRoomIndex lecturesByRoomIndex = new LecturesByRoomIndex(lectures);
         LayoutCalculator layoutCalculator = new LayoutCalculator(Logging.Companion.get(), boxHeight);
 
-        for (int columnIndex = 0; columnIndex < MyApp.roomCount; columnIndex++) {
+        for (int columnIndex = 0; columnIndex < roomCount; columnIndex++) {
```

Comment:
It feels painful to have UI code map between column index and room index, and also having to look up which lectures belong to a room.

Wouldn't this all be easier by passing a `List<RoomDay>` to the UI code. Something like this:

```kotlin
data class RoomDay(
    val day: Whatever,
    val room: Room,
    val lectures: List<Lecture>
)

data class Room(
    // Personally, I would make this the database ID and only refer to rooms by this 
    val roomIndex: Int, 
    val name: String
)
```

Reply:
Agreed. I plan to restructure the data types and wrap related fields in a class. - At this point I like to focus on the decoupling step.